lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -4228,7 +4228,8 @@ class SROA_Parameter_HLSL : public ModulePass {
 
       // Skip library function, except to LegalizeDxilInputOutputs
       if (&F != m_pHLModule->GetEntryFunction() &&
-          !m_pHLModule->IsEntryThatUsesSignatures(&F)) {
+          !m_pHLModule->IsEntryThatUsesSignatures(&F) &&
+          !m_pHLModule->IsNodeShader(&F)) {
```

Comment:
This change doesn't go along with the PR title of "Mark Node as EntryThatUsesSignature for HLModule", and that title doesn't indicate what's actually been fixed either.  What about "Enable flattening of node non-object parameters" (if that's accurate)?

Can node actually be included in `EntryThatUsesSignature`, or will the one remaining use in HLMatrixLower blow up?  Can it be described why we include node in some places that call `IsEntryThatUsesSignatures` and not in another?

Also, There is a corresponding method with the same name on `DxilModule` which does include node and has different usages.  I think the names of the methods `EntryThatUsesSignature` need updating, since the meaning and purpose is not the same for each.

Reply:
Or "Flatten non-object parameters for node shaders"